python/lsst/pipe/base/task.py

```diff
@@ -111,7 +111,7 @@ def __init__(self, config=None, name=None, parentTask=None, log=None):
             if config is None:
                 config = getattr(parentTask.config, name)
             self._taskDict = parentTask._taskDict
-            self.log = Log.getLogger(parentTask.log.getName() + '.' + name)
+            loggerName = parentTask.log.getName() + '.' + name
```

Comment:
@hchiang2 Does the new `log` package have the concept of child logs that share the same stream as a parent log, like the older `pex_logging` package? If so, then you have lost that inheritance with your changes and should revisit the code. If the new `log` package has no such inheritance then your changes are safe.

The documentation for the `log` argument is weak, which is my fault, but I'd ask you to see if you can clean it up here (or I'll take a pass at it if you prefer). "The full task name is used..." is not informative. Also, if your changes are correct then the documentation would be improved by stating that the `log` argument is only used for its name.